pkg/transport/bprotocol/compute_proxy.go

```diff
@@ -146,5 +149,60 @@ func proxyRequest[Request any, Response any](
 	return result.Rehydrate()
 }
 
-// Compile-time interface check:
+func proxyStreamingRequest[Request any, Response any](
+	ctx context.Context,
+	h host.Host,
+	destPeerID string,
+	protocolID protocol.ID,
+	request Request) (<-chan *concurrency.AsyncResult[Response], error) {
+	// decode the destination peer ID string value
+	peerID, err := peer.Decode(destPeerID)
+	if err != nil {
+		return nil, fmt.Errorf("%s: failed to decode peer ID %s: %w", reflect.TypeOf(request), destPeerID, err)
+	}
+
+	// deserialize the request object
+	data, err := json.Marshal(request)
+	if err != nil {
+		return nil, fmt.Errorf("%s: failed to marshal request: %w", reflect.TypeOf(request), err)
+	}
+
+	// opening a stream to the destination peer
+	stream, err := h.NewStream(ctx, peerID, protocolID)
+	if err != nil {
+		return nil, fmt.Errorf("%s: failed to open stream to peer %s: %w", reflect.TypeOf(request), destPeerID, err)
+	}
+	if scopingErr := stream.Scope().SetService(ComputeServiceName); scopingErr != nil {
+		_ = stream.Reset()
+		return nil, fmt.Errorf("%s: failed to attach stream to compute service: %w", reflect.TypeOf(request), scopingErr)
+	}
+
+	// write the request to the stream
+	_, err = stream.Write(data)
+	if err != nil {
+		_ = stream.Reset()
+		return nil, fmt.Errorf("%s: failed to write request to peer %s: %w", reflect.TypeOf(request), destPeerID, err)
+	}
+
+	ch := make(chan *concurrency.AsyncResult[Response])
+	go func() {
+		defer stream.Close() //nolint:errcheck
+		defer close(ch)
+		for {
+			response := new(concurrency.AsyncResult[Response])
+			err = json.NewDecoder(stream).Decode(response)
+			if err != nil {
+				_ = stream.Reset()
+				if err != io.EOF {
+					response.Err = fmt.Errorf("%s: failed to decode response from peer %s: %w", reflect.TypeOf(request), destPeerID, err)
```

Comment:
`%T` should work instead of reflect.

Reply:
Fixed in https://github.com/bacalhau-project/bacalhau/pull/3354/commits/80d1eef8857a74bd9ceb967d9958041bbcd42308